Convert test

jens-erdmann opened the pull request:
Convert existing bash tests into pytest. See commits for details.

jens-erdmann committed:
tests: Introduce skeletons for new tests

Introduce new test files for compatibilty checks and outbound
candidates. Theses tests should replace their bash counterparts short
term.
Both tests are just skeltons yet due to missing pytest components.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

priv-kweihmann committed:
setup: add pytest-forked

to fork a new clean process per test run

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>

priv-kweihmann committed:
tests: reenable disabled tests

- use json.loads to turn lists back into comparable objects

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>

jens-erdmann committed:
test_outbounds: Convert existing tests

Convert existing tests of outbounds.sh into python tests.
One test case is disabled due to an error.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
display-compat: Remove Test

Remove test file that was replicated in pytest already.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
test_simplify: Add skeleton to test simplify

Add skeleton for a test to implement tests for the simplification
function of flict.
This test will replace license_expr.sh short term.
It is a skeleton for now as there are issues regarding results.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

priv-kweihmann commented on tests/test_compat.py:
```diff
@@ -0,0 +1,40 @@
+import sys
+import unittest
+
+TEST_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```
As mentioned earlier already, I would like this sys.path manipulation to be move into a class function

```python
@classmethod
def setup_class(cls):
    sys.path.insert(0, os.path.abspath(os.path.dirname(__file__) + "/../"))

    try:
        from flict.flictlib import flict_config
        from flict.impl import FlictImpl
    except ImportError as e:
        self.fail("Can't import flict... {msg}".format(e))
```

jens-erdmann replied:
Should be fixed by now.

priv-kweihmann reviewed (changes requested):
Please update the sys.path manipulation, otherwise I'm good with this change - definitely this is a move in the right direction

jens-erdmann committed:
tests: Inroduce Mock for Argparse

Introduce mock class for argparse. There was a mock already implemented
in outbounds, compat and simplified tests. These are now replaced with
the shared class.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
tests: Remove unneeded imports and env settings

Remove imports and environment settings that are not needed for the new
tests.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
requirements-dev: Add dataclasses

Add missing dataclasses package in the requirements to enable usage of
ArgsMock class.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
tests: Convert New Tests to Pytest

Convert newly introduced tests to pytest to avoid manually enforcing
local imports

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
test_outbounds: Fix test

Remove relicense file in args mock to mimik the same behavoir as in the
main class.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
outbounds: Remove bash test

Remove outbounds tests written in bash. This file has been converted
into pytest.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
test_simplify: Fix test

Fix copy past error in helper function. Enable tests to mimik tests of
license-expr.sh.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
license-expr: Remove bash test

Remove license-expr test written in bash. This test has now been
converted to pytest.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

priv-kweihmann commented on tests/test_outbounds.py:
```diff
@@ -0,0 +1,41 @@
+from flict.impl import FlictImpl
+from tests.args_mock import ArgsMock
+
+def _test_expression(expression, result):
```
Can we move that function to a common class/file? I already see code duplication here

jens-erdmann replied:
Not really because it calls a different method of FlictImpl in each test.
